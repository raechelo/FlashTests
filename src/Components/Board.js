import React, { Component } from 'react';
import PlayerScore from './PlayerScore';
import Card from './Card';


export default class Board extends Component {
  constructor(props) {
    super(props);
    this.state = {
      questions: [],
      currQ: {},
      answerBank: [],
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps !== this.props) {
      this.setState( {questions: this.props.questions} )
    }
  }

  shuffleQs = () => {
    let currentQ = this.state.questions.sort(() => 0.5 - Math.random()).pop()
    this.setState( {currQ: currentQ} )
  }

  shuffleAs = () => {
    let ans = this.state.questions.reduce((acc, q) => {
      acc.push(q.answer)
      return acc
    }, [])
    let shuffledAs = ans.sort(() => 0.5 - Math.random())
    this.setState( {answerBank: shuffledAs} )
  }

  render() {
    return (
        <div className="card-board" >
          <Card q={this.state.currQ} 
          nextQ={this.shuffleQs} 
          ansBank={this.state.answerBank} 
          iq={this.props.iq}
          startG={this.shuffleAs}
          saveToStore={this.props.saveToStore}
          oldCards={this.oldCards}
          />
          <PlayerScore />
        </div>
    )
  }
}