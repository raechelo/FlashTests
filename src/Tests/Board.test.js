import React from 'react';
import Board from '../Components/Board';
import { shallow } from 'enzyme';

describe('Board', () => {

  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(
      <Board />
    )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the proper default state', () => {
    expect(wrapper.state()).toEqual( {
      questions: [], currQ: {}, answerBank: []
    })
  });
})