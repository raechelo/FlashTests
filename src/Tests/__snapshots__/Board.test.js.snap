// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should match the snapshot with all the data passed in 1`] = `
<div
  className="card-board"
>
  <Card
    ansBank={Array []}
    nextQ={[Function]}
    q={Object {}}
    startG={[Function]}
  />
  <PlayerScore />
</div>
`;
